### PR TITLE
ARROW-9620: [C++] Added split function to Gandiva

### PR DESCRIPTION
**Pull request by 18dcb01**

### CODE REVIEW

**Comment by emkornfield** [comment]
@praveenbingo @projjal @pravindra could one of you take a look?

**Review by projjal — APPROVED** [review]

File: cpp/src/gandiva/precompiled/string_ops.cc [hunk 1]
```diff
@@ -567,6 +567,54 @@ const char* castVARCHAR_utf8_int64(gdv_int64 context, const char* data,
   return data;
 }
 
+// Split a string into substrings, then return a particular one
+FORCE_INLINE
+const char* split_utf8_utf8_int32(gdv_int64 context, const char* str, gdv_int32 str_len,
+                                  const char* delimiter, gdv_int32 delimiter_len,
+                                  gdv_int32 ret_index, gdv_int32* out_len) {
+  if (ret_index < 1) {
+    gdv_fn_context_set_error_msg(context, "Substring index must be greater than 0");
```
Comment (emkornfield) [inline cpp/src/gandiva/precompiled/string_ops.cc:576]: does this match a specific sql function?  otherwise I would expect 0 based indexing?

**Comment by emkornfield** [comment]
+1 merging.

**Comment by assignUser** [comment]
@emkornfield this seems to have gotten lost/overlooked, should we keep the PR open?

**Comment by emkornfield** [comment]
Yeah, I don't know why I didn't actually merge it.  I think it is fine to close, as there hasn't been follow-up.

**Pull request closed without merging**